src/app/models/event.model.spec.ts

```diff
@@ -142,4 +225,184 @@ describe('Unit Test :: Events Model', () => {
     };
     expect(expectedOutPut).toEqual(output);
   });
+
+  it('should correctly convert to ui model - 5', () => {
+    const map = new EventMapper();
+    const input = {
+      'attributes': {
+        'name': 'system.assignees',
+        'timestamp': '2018-05-27T08:54:44.63509Z'
+      },
+      'id': '5e7adc1d-2c8a-440d-9466-aedf4b2d60c2',
+      'relationships': {
+        'modifier': {
+          'data': {
+            'id': '20694424-0841-4d6c-bfb5-bbbb0391b8db',
+            'links': {
+              'related': 'https://api.prod-preview.openshift.io/api/users/20694424-0841-4d6c-bfb5-bbbb0391b8db'
+            },
+            'type': 'identities'
+          }
+        },
+        'newValue': {
+          'data': [
+            {
+              'id': '20694424-0841-4d6c-bfb5-bbbb0391b8db',
+              'type': 'users'
+            }
+          ]
+        },
+        'oldValue': {
+          'data': [
+            {
+              'id': '20694424-0841-4d6c-bfb5-bbbb0391b8db',
+              'type': 'users'
+            }
+          ]
+        },
+        'workItemType': {
+          'data': {
+            'id': '00000000-0000-0000-0000-000000000003',
+            'type': 'workitemtypes'
+          },
+          'links': {
+            'self': 'http:///api/workitemtypes/00000000-0000-0000-0000-000000000003'
+          }
+        }
+      },
+      'type': 'events'
+    };
+    const output: EventUI = map.toUIModel(input);
+    const expectedOutPut: EventUI = {
+      name: 'system.assignees',
+      newValue: null,
+      oldValue: null,
+      timestamp: '2018-05-27T08:54:44.63509Z',
+      modifierId: '20694424-0841-4d6c-bfb5-bbbb0391b8db',
+      newValueRelationships: [
+        {
+          id: '20694424-0841-4d6c-bfb5-bbbb0391b8db',
+          type: 'users'
+        }
+      ],
+      oldValueRelationships: [
+        {
+          id: '20694424-0841-4d6c-bfb5-bbbb0391b8db',
+          type: 'users'
+        }
+      ],
+      type: null
```

Comment:
It seems like the work item type from the payload (`00000000-0000-0000-0000-000000000003`) is not parsed here but you need this information.

Reply:
@kwk type is not parsed here but on a later stage, before events are stored in `ngrx/store`.

Reply:
@sahil143 okay fine as long it is taken the work item type that is given by the event that is fine. When rendering events you must know at each point in time what the type of the work item was at that given time. In the near future @jarifibrahim will introduce type changes and then you might have to do a lookup to find out what field was actually changed.